### ISSUE TITLE
ALTER PROCEDURE is not supported

### ISSUE THREAD

**Issue by moreaupascal56** [sqlfluff/sqlfluff#5269]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

As discussed in [3479](https://github.com/sqlfluff/sqlfluff/issues/3479), any ALTER PROCEDURE query is not handled by SQLFLUFF and gives error (`Found unparsable section`). 

All the queries below fails (and are from snowflake examples from the [doc](https://docs.snowflake.com/en/sql-reference/sql/alter-procedure)): 

```sql
ALTER PROCEDURE test(int) RENAME TO newname;
ALTER PROCEDURE test(int) SET COMMENT = 'string_literal';
ALTER PROCEDURE test(int) UNSET COMMENT;
ALTER PROCEDURE test(int)  SET TAG tag_name = 'v'  ,tag_name2 = 'v';
ALTER PROCEDURE test(int)  UNSET  TAG tag_name  ,tag_name2;
ALTER PROCEDURE test(int)  EXECUTE AS CALLER;
 ALTER PROCEDURE test(int)  EXECUTE AS OWNER;

```

I may submit a PR but if anyone wants to do it before feel free :D 

### Expected Behaviour

Lint ALTER PROCEDURE queries

### Observed Behaviour

Found unparsable section

### How to reproduce

Run the queries mentioned above

### Dialect

snowflake

### Version

2.3.2

### Configuration

Just running with `sqlfluff lint --dialect snowflake`. 

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by moreaupascal56** [comment]
in fact we are working well with the original issue, closing this :)
